### PR TITLE
Improve Read mode

### PR DESCRIPTION
**Pull request by elland**

### CODE REVIEW

**Comment by 3lvis** [comment]
Is almost like every PR had a conflict 🤔

**Comment by 3lvis** [comment]
Nevermind, it was just me.

**Comment by 3lvis** [comment]
Damn. 

screen shot 2016-10-26 at 11 02 55 am

**Comment by elland** [comment]
Feels good deleting so much code. This PR is sadly almost unreadable because of all the Xcode noise.

**Comment by 3lvis** [comment]
@Elland wasn't too hard to parse the changes in this PR.

**Comment by 3lvis** [comment]
Great work! Ship it.

**Comment by elland** [comment]
💃

**Comment by elland** [comment]
Yeah mostly noise.
